### PR TITLE
Support mapping team-members to SCIM id's in order to align with userIds used in auth

### LINKED ISSUE #111: gh provider, map users via scim username
Our usecase:
1. we use AAD OIDC for user auth in OCP (using upn as userid)
2. we do not auth with GH as OIDC provider, as we want users to have singular identities, and the user-audience of OCP is wider than GH.
2. we have various teams in GH, which would map nicely over to groups in OCP and hence roles

However, syncing GH teams as groups with the current impl. would use the gh login id, while we'd like to list team-members, find their linked SCIM-identity, and then use the `Username` attribute, which would match upn.

Would you accept such an implementation? Could fit into the same package, and be toggled via a flag?

### PR DESCRIPTION
**Pull request by davidkarlsen**
Fixes #111.
Currently a draft as docs are needed to add, and also there is a snag with app-based auth when using the v4 client (https://github.com/palantir/go-githubapp/issues/104).

### CODE REVIEW

**Comment by davidkarlsen** [comment]
@sabre1041 PTAL.

This makes it possible to map teams from github, resolving the userId to the IDP-identity. This way the github-teams member uids will align with the id used in the IDP when using AAD as IDP. Use-case outlined in #111.

Note that this introduces a couple of fields in the CRD when using GH as provider.

**Comment by davidkarlsen** [comment]
@sabre1041 PTAL?

File: README.md [hunk 1]
```diff
@@ -178,7 +178,7 @@ When authenticating as a Github App, the following keys are required:
 In GitHub, go to developer-settings -> github apps.
 * Create a new app, it does not need webhook callbacks.
 * Generate a private-key and download it
-* Under "permissions and events" the app will need read-only access to the "Members" permission in the "Organization" section.
+* Under "permissions and events" the app will need read-only access to the "Members" permission in the "Organization" section. NOTE: If you enable `MapByScimId` this permissions needs to be _Read & Write_, although, in-fact the operator only does read-only operations. The reason for this is the use of the v4 graphql api-endpoint.
```
Comment (sabre1041) [inline README.md:181]:
```suggestion
* Under "permissions and events", the app will need read-only access to the "Members" permission in the "Organization" section. NOTE: If you enable `mapByScimId`, this permissions needs to be _Read & Write_, though the operator only does read-only operations. The reason for this is the use of the v4 graphql api-endpoint.
```

**Review by sabre1041 — CHANGES_REQUESTED** [review]
Small docs change.

I do not have a SCIM endpoint, so my testing later today will be limited to ensuring that the existing GitHub functionality is retained

**Review by sabre1041 — APPROVED** [review]
LGTM

**Comment by davidkarlsen** [comment]
> LGTM

Awesome - when (ca) will the next release be?

**Pull request merged**